frontend/amundsen_application/static/js/pages/SearchPage/index.tsx

```diff
@@ -132,14 +114,16 @@ export class SearchPage extends React.Component<
   };
 
   getTabContent = (results: SearchResults<Resource>, tab: ResourceType) => {
-    const { didApplyFilters } = this.state;
-    const { hasFilters, searchTerm, setPageIndex } = this.props;
+    const { hasFilters, searchTerm, setPageIndex, didSearch } = this.props;
     const { page_index, total_results } = results;
     const startIndex = RESULTS_PER_PAGE * page_index + 1;
     const tabLabel = this.generateTabLabel(tab);
 
-    // No search input
-    if (searchTerm.length === 0 && (!hasFilters || !didApplyFilters)) {
+    // No search input and search action has not been taken
```

Comment:
I would rather read this as the name of a variable, something like:
```const hasNoSearchInputOrAction = searchTerm.length === 0 &&
      (!hasFilters || (!didSearch && total_results === 0));
      ```
      
      or something similar